### DISCUSSION TITLE
2023-10-21 Discussion: Broker overall interface, Q&A, Updates

### DISCUSSION THREAD

**Discussion by isaisa002** [Broke-Brokers/soen341_fproject2023#121] [Q&A] [answered]
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/118073058/191c1419-9f5e-4b43-ba2e-76cc9ae3b5d7)

![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/118073058/79cc503b-9041-45d1-916a-2091cbc30fd4)

## discuss #34 #57 
1. For now, Search property page for broker is the same as the one for homeowner. Does brokers & home owner/renter needs different Property page, request page ?
2. Should we make a page for save as favourite to clearly show that home owner/renter saves into their account and broker save into client's profile ?
3. Need database for requests form, I volunteer to make it (need to create issues related)

## discuss #65 
4. explain field created and see if respects user stories
5. Discuss: https://www.pluralsight.com/guides/load-an-existing-database-record-to-a-react.js-form 

## discuss #37 
6. system admin mockup and functionalities

## General
7. In acceptance criteria for a user story, documentation for every components is needed. Documentation should be only comments of this issue or need a pdf that explains components on repository ?
8. In acceptance criteria for a user story, Cypress acceptance test in GIF form is needed what's that ?

Cypress X 
Github Workflow YES

•	GitHub actions QuickStart. https://docs.github.com/en/actions/quickstart
•	Building and testing Java with Ant. https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-ant 
•	Workflow syntax for GitHub Actions. https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions  

9. Task breakdown startegy: User story for big thing, creates new issues as task description from task breakdown?

**Comment by DominiqueProulx** [comment, marked as answer]
discusss #34 #57


> 1.For now, Search property page for broker is the same as the one for homeowner. Does brokers & home owner/renter needs different Property page, request page ?
I went in #34, seems like the only thing that changes is that brokers have a "send to client " button and saved to clients file. 

I would suggest that , for simplicity , we move these two to "property page" and that it works with a logic test on who's logged in (homebuyer or broker)  What do you all think? 
Also we don't need those buttons (save to clients + ) it in sprint #2 we could create the task but leave it for a later sprint. 


>2. Should we make a page for save as favourite to clearly show that home owner/renter saves into their account and broker save into client's profile ?

It can a section in the homebuyer/renter profile  + brokers client profile page with a gallery of saved properties (component could be reused)? 
This would be for another sprint. It does not appear in sprint #2 or sprint #3 criteria (thank god)


>3. Need database for requests form, I volunteer to make it (need to create issues related)

We could just use a mailto protocol (when we send the form it sends an email directly )since the broker's database have a email field? would save us a lot of time.

---

general 

> In acceptance criteria for a user story, documentation for every components is needed. Documentation should be only comments of this issue or need a pdf that explains components on repository ?

No need for more documents, the issue is enough 

> In acceptance criteria for a user story, Cypress acceptance test in GIF form is needed what's that ?

Cypress is a tool used to simulate/test user interaction and record it. As I understand now they just want us to set up the tests with github actions. Like in Devops Slides p .  25 to 29

> Task breakdown strategy: User story for big thing, creates new issues as task description from task breakdown?

Yess!

1.User story are upper level features that need to be implemented during a sprint 
2.User story is broken down into pieces : the task breakdown 
3. we create tasks from the task breakdown and work on that bit at a time.

**Comment by DominiqueProulx** [comment]
2023-10-21 UPDATE



## Important
I realises that we took the high level user stories and thought that we needed to implement almost all the features. 
When they are lisiting the features to implement for next sprint, they are referring to the ordered list from the project description and sprint 1 description.  This cuts down out work but we are still behind. but those are good news.

We should re-organize the user stories a little bit to reflect this. I'll come up with a plan, discussions are welcome . The task breakdown for sprint3 that I am about to do will reflect this.  Questions/comments? 

---

## From the Sprint 2 details document  1. Develop and implement each feature



 These are the features we need to prioritize now if we want to be able to get the sprint2 points 
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/ab239f1f-dce3-439b-a4c9-553719367888)


### a) Search for properties
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/01b19675-49db-464e-b4f3-a7cb39fcbd36)

SEARCH PAGE

this feature is accessed from main navigation bar

- search bar #69 #96 
- result gallery #69 #96 
- connect to property database (Rebecca was able to make react communicate with database!! ) #25 #60  
- property page (visual done but not perfect) #60 #97 
  - needs the field connected to database #25 

### b) Request a visit to property
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/ac796e3a-9729-49c1-8307-c22b76cf7659)

this feature is accessed through the property page. 

- property page button (done) #117 
- form accessed from property page or request page : discussion in progress #72 

### c) CRUD operations on properties
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/e53ee46f-8976-4174-b02b-bf4038a06738)

this feature is accessed through the the brokers profile page

- login #68 
- brokers profile page #34
  - House offer section (brokers property listing section)(no task created yet) 
       -  add button component created #119 
       - delete button
       - edit button : goes to a form
       - properties cards 
  - property card ? 
  - form to add property
  - link to database #25   
     -  link brokers to their lisiting
     -  link "add" "edit " " delete"  

### d) CRUD operations on brokers 
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/9b794fcc-edca-4dd5-a37f-17ee1d777c4f)

- sysadmin page (created) #user story #37
- task breakdown in issue #37 
- grid layout with brokers cards
       -  add button component created #119 
       - delete button
       - edit button : goes to a form
- link to brokers database #25 #63 
- form to edit brokers info #63 
- form to add a broker #63 

---

## From the Sprint 2 details document (con'd)
  2. Write 2 acceptance tests

![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/c71c76ef-bbd4-4764-b3fb-099f89d81d89)


3. Set up your Continuous Integration pipeline in your repository. 
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/25d7b79d-b565-46ac-902c-e09527a70ff9)

  #111 + let's continue discussion from above.

4. Plan for the next sprint the same way that you did for Sprint 2.  
Thats what i'll do this week. 
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/dae3d5a9-b889-4443-b133-174836acfe2c)

5.6.7.8
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/39106966-89db-4e9e-b004-a457808a4722)
